replica-node/src/replica_node.rs

```diff
@@ -0,0 +1,318 @@
+use {
+    crossbeam_channel::unbounded,
+    log::*,
+    solana_download_utils::download_snapshot,
+    solana_genesis_utils::download_then_check_genesis_hash,
+    solana_gossip::{cluster_info::ClusterInfo, contact_info::ContactInfo},
+    solana_ledger::{
+        blockstore::Blockstore, blockstore_db::AccessType, blockstore_processor,
+        leader_schedule_cache::LeaderScheduleCache,
+    },
+    solana_rpc::{
+        max_slots::MaxSlots,
+        optimistically_confirmed_bank_tracker::{
+            OptimisticallyConfirmedBank, OptimisticallyConfirmedBankTracker,
+        },
+        rpc::JsonRpcConfig,
+        rpc_pubsub_service::{PubSubConfig, PubSubService},
+        rpc_service::JsonRpcService,
+        rpc_subscriptions::RpcSubscriptions,
+    },
+    solana_runtime::{
+        accounts_index::AccountSecondaryIndexes,
+        bank_forks::BankForks,
+        commitment::BlockCommitmentCache,
+        hardened_unpack::MAX_GENESIS_ARCHIVE_UNPACKED_SIZE,
+        snapshot_config::SnapshotConfig,
+        snapshot_utils::{self, ArchiveFormat},
+    },
+    solana_sdk::{clock::Slot, exit::Exit, genesis_config::GenesisConfig, hash::Hash},
+    std::{
+        fs,
+        net::SocketAddr,
+        path::PathBuf,
+        sync::{
+            atomic::{AtomicBool, AtomicU64},
+            Arc, RwLock,
+        },
+    },
+};
+
+pub struct ReplicaNodeConfig {
+    pub rpc_source_addr: SocketAddr,
+    pub rpc_addr: SocketAddr,
+    pub rpc_pubsub_addr: SocketAddr,
+    pub ledger_path: PathBuf,
+    pub snapshot_output_dir: PathBuf,
+    pub snapshot_path: PathBuf,
+    pub account_paths: Vec<PathBuf>,
+    pub snapshot_info: (Slot, Hash),
+    pub cluster_info: Arc<ClusterInfo>,
+    pub rpc_config: JsonRpcConfig,
+    pub snapshot_config: Option<SnapshotConfig>,
+    pub pubsub_config: PubSubConfig,
+    pub account_indexes: AccountSecondaryIndexes,
+    pub accounts_db_caching_enabled: bool,
+    pub replica_exit: Arc<RwLock<Exit>>,
+}
+
+impl Default for ReplicaNodeConfig {
+    fn default() -> Self {
+        Self {
+            rpc_source_addr: SocketAddr::from(([127, 0, 0, 1], 8001)),
+            rpc_addr: SocketAddr::from(([127, 0, 0, 1], 8001)),
+            rpc_pubsub_addr: SocketAddr::from(([127, 0, 0, 1], 8001)),
```

Comment:
```suggestion
            rpc_source_addr: socketaddr_any!(),
            rpc_addr: socketaddr_any!(),
            rpc_pubsub_addr: socketaddr_any!(),
```
In case we ever use this default for tests?

Reply:
Cool. Thanks